Octokit/Clients/RepositoryCommitsClient.cs

```diff
@@ -105,6 +107,74 @@ public Task<CompareResult> Compare(long repositoryId, string @base, string head)
             return ApiConnection.Get<CompareResult>(ApiUrls.RepoCompare(repositoryId, @base, head));
         }
 
+        /// <summary>
+        /// Compare two references in a repository
+        /// </summary>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="base">The reference to use as the base commit</param>
+        /// <param name="head">The reference to use as the head commit</param>
+        /// <param name="options">Options for changing the API response</param>
+        [ManualRoute("GET", "/repos/{owner}/{repo}/compare/{base}...{head}")]
+        public Task<CompareResult> Compare(string owner, string name, string @base, string head, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNullOrEmptyString(@base, nameof(@base));
+            Ensure.ArgumentNotNullOrEmptyString(head, nameof(head));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            return Compare(ApiUrls.RepoCompare(owner, name, @base, head), options);
+        }
+
+        /// <summary>
+        /// Compare two references in a repository
+        /// </summary>
+        /// <param name="repositoryId">The Id of the repository</param>
+        /// <param name="base">The reference to use as the base commit</param>
+        /// <param name="head">The reference to use as the head commit</param>
+        /// <param name="options">Options for changing the API response</param>
+        [ManualRoute("GET", "/repositories/{id}/compare/{base}...{head}")]
+        public Task<CompareResult> Compare(long repositoryId, string @base, string head, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(@base, nameof(@base));
+            Ensure.ArgumentNotNullOrEmptyString(head, nameof(head));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            return Compare(ApiUrls.RepoCompare(repositoryId, @base, head), options);
+        }
+
+        private async Task<CompareResult> Compare(Uri uri, ApiOptions options)
+        {
+            var results = await ApiConnection.GetAll<CompareResult>(uri, options);
+            if (results.Count == 1) return results[0];
+
+            var firstCompareResult = results[0];
+            var commits = firstCompareResult.Commits.ToList();
+            var files = firstCompareResult.Files.ToList();
+
+            foreach (var compareResult in results.Skip(1))
```

Comment:
I'm sorry I'm blanking on this: do you mind explaining to me why we're skipping a result here?

Reply:
Oh of course, :facepalm:. Thanks.